zcash_primitives/src/zip32.rs

```diff
@@ -233,6 +235,41 @@ pub fn sapling_default_address(
     sapling_find_address(fvk, dk, DiversifierIndex::new()).unwrap()
 }
 
+/// Returns the internal full viewing key and diversifier key
+/// for the provided external fvk and dk
+pub fn sapling_derive_internal_fvk(
+    fvk: &FullViewingKey,
+    dk: &DiversifierKey,
+) -> (FullViewingKey, DiversifierKey) {
+    let i = {
+        let mut h = Blake2bParams::new()
+            .hash_length(64)
```

Comment:
ZIP 32 specifies `BLAKE2b-256` here:
```suggestion
            .hash_length(32)
```
This bug should be caught by test vectors.